packages/components/tree-select/src/tree.ts

```diff
@@ -120,6 +121,30 @@ export const useTree = (
     })
   }
 
+  useEventListener(
+    select,
+    'keydown',
+    (e) => {
+      const { code } = e
+
+      if (
+        (code !== EVENT_CODE.enter && code !== EVENT_CODE.numpadEnter) ||
+        props.checkStrictly
+      ) {
+        return
+      }
+
+      const { hoveringIndex } = select.value?.states ?? {}
+      const option = select.value?.optionsArray[hoveringIndex]
+      const { isLeaf } = tree.value?.getNode(option?.value) ?? {}
+
+      if (!isLeaf) {
+        e.stopPropagation()
+      }
+    },
+    { capture: true }
+  )
```

Comment:
Thanks you for your contribution. I think a more complete solution is to refine the logic of the `enter` event based on the `click` event of the current tree-select-option.